samples/attested_tls/server/enc/openssl_server.cpp

```diff
@@ -0,0 +1,200 @@
+{
+    int ret = -1;
+
+waiting_for_connection_request:
+
+    // reset ssl_session and client_socket_fd to prepare for the new TLS
+    // connection
+    close(client_socket_fd);
+    SSL_free(ssl_session);
+    printf(TLS_SERVER " waiting for client connection\n");
+
+    struct sockaddr_in addr;
+    uint len = sizeof(addr);
+    client_socket_fd = accept(server_socket_fd, (struct sockaddr*)&addr, &len);
+
+    if (client_socket_fd < 0)
+    {
+        printf(TLS_SERVER "Unable to accept the client request\n");
+        goto exit;
+    }
+
+    // create a new SSL structure for a connection
+    if ((ssl_session = SSL_new(ssl_server_ctx)) == nullptr)
+    {
+        printf(TLS_SERVER
+               "Unable to create a new SSL connection state object\n");
+        goto exit;
+    }
+
+    SSL_set_fd(ssl_session, client_socket_fd);
+
+    // wait for a TLS/SSL client to initiate a TLS/SSL handshake
+    if (SSL_accept(ssl_session) <= 0)
+    {
+        printf(TLS_SERVER " SSL handshake failed\n");
+        goto exit;
+    }
+
+    printf(TLS_SERVER "<---- Read from client:\n");
+    if (read_from_session_peer(
+            ssl_session, CLIENT_PAYLOAD, CLIENT_PAYLOAD_SIZE) != 0)
+    {
+        printf(TLS_SERVER " Read from client failed\n");
+        goto exit;
+    }
+
+    printf(TLS_SERVER "<---- Write to client:\n");
+    if (write_to_session_peer(
+            ssl_session, SERVER_PAYLOAD, strlen(SERVER_PAYLOAD)) != 0)
+    {
+        printf(TLS_SERVER " Write to client failed\n");
+        goto exit;
+    }
+
+    if (keep_server_up)
+        goto waiting_for_connection_request;
+
+    ret = 0;
+exit:
+    return ret;
+}
+
+int set_up_tls_server(char* server_port, bool keep_server_up)
+{
+    int ret = 0;
+    int server_socket_fd;
+    int client_socket_fd;
+    int server_port_number;
+
+    X509* certificate = nullptr;
+    EVP_PKEY* pkey = nullptr;
+    SSL_CONF_CTX* ssl_confctx = SSL_CONF_CTX_new();
+
+    SSL_CTX* ssl_server_ctx = nullptr;
+    SSL* ssl_session = nullptr;
+
+    /* Load host resolver and socket interface modules explicitly */
+    if (load_oe_modules() != OE_OK)
+    {
+        printf(TLS_SERVER "loading required Open Enclave modules failed\n");
+        goto exit;
+    }
+
+    if ((ssl_server_ctx = SSL_CTX_new(TLS_server_method())) == nullptr)
+    {
+        printf(TLS_SERVER "unable to create a new SSL context\n");
+        goto exit;
+    }
+
+    if (initalize_ssl_context(ssl_confctx, ssl_server_ctx) != OE_OK)
+    {
+        printf(TLS_SERVER "unable to create a initialize SSL context\n ");
+        goto exit;
+    }
+
+    SSL_CTX_set_verify(ssl_server_ctx, SSL_VERIFY_PEER, &verify_callback);
+
+    if (load_tls_certificates_and_keys(ssl_server_ctx, certificate, pkey) != 0)
+    {
+        printf(TLS_SERVER
+               " unable to load certificate and private key on the server\n ");
+        goto exit;
+    }
+
+    sscanf(server_port, "%d", &server_port_number); // convert to char* to int
```

Comment:
Can you use `atoi` instead? `sscanf` causes some memory issues in the tls_e2e test.